subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultCopySpec.java

```diff
@@ -662,16 +663,17 @@ public RelativePath getDestPath() {
                 parentPath = parentResolver.getDestPath();
             }
 
-            String path = DefaultCopySpec.this.getDestPath();
+            File path = DefaultCopySpec.this.getDestinationDir();
             if (path == null) {
                 return parentPath;
             }
 
-            if (path.startsWith("/") || path.startsWith(File.separator)) {
-                return RelativePath.parse(false, path);
+            String pathAsString = path.getPath();
+            if (pathAsString.startsWith("/") || pathAsString.startsWith(File.separator)) {
+                return RelativePath.parse(false, pathAsString);
             }
 
-            return RelativePath.parse(false, parentPath, path);
+            return RelativePath.parse(false, parentPath, pathAsString);
```

Comment:
🤔 This looks all sorts of weird.

`path.getPath` is not quite the same thing as `getDestPath` was returning before.  We're turning `destPath` into a `File` and then going back to a `String` to manipulate it more.

Can we avoid tracking the destination directory completely for now?

I think saying:
> Cannot copy zip entry '/Users/sterling/gits/gradle/subprojects/core/build/tmp/teŝt files/CopyTaskInt.Spec/8mt32/before/files.zip!sub/c.txt' to 'other/sub/c.txt' because file '/Users/sterling/gits/gradle/subprojects/core/build/tmp/teŝt files/CopyTaskInt.Spec/8mt32/before/files/sub/c.txt' has already been copied there.

is already a big improvement and would make the error consistent across all Copy-like tasks (Copy, Jar, etc)

Reply:
I reverted all this and restored the original `getDestPath()` method.  I was just trying to consolidate near duplicate methods here - the weirdness exists prior to this change.

I think this can keep the destination dir for the message with this change.  I think it makes a clearer message to provide it if at all possible.